src/pages/ReimbursementAccount/ACHContractStep.js

```diff
@@ -114,45 +153,38 @@ class ACHContractStep extends React.Component {
             || (_.size(this.state.beneficialOwners) === 3 && !this.state.ownsMoreThan25Percent);
     }
 
-    /**
-     * Clear the error associated to inputKey if found and store the inputKey new value in the state.
-     *
-     * @param {Integer} ownerIndex
-     * @param {Object} values
-     */
-    clearErrorAndSetBeneficialOwnerValues(ownerIndex, values) {
-        this.setState((prevState) => {
-            const beneficialOwners = [...prevState.beneficialOwners];
-            beneficialOwners[ownerIndex] = {...beneficialOwners[ownerIndex], ...values};
-            BankAccounts.updateReimbursementAccountDraft({beneficialOwners});
-            return {beneficialOwners};
-        });
-
-        // Prepare inputKeys for clearing errors
-        const inputKeys = _.keys(values);
-
-        // dob field has multiple validations/errors, we are handling it temporarily like this.
-        if (_.contains(inputKeys, 'dob')) {
-            inputKeys.push('dobAge');
+    submit(values) {
+        if (!this.validate(values)) {
+            return;
         }
-        this.clearErrors(_.map(inputKeys, inputKey => `beneficialOwnersErrors.${ownerIndex}.${inputKey}`));
-    }
 
-    submit() {
-        if (!this.validate()) {
-            return;
+        // Because we do not update the state of the form, we must filter removed beneficial owners and reformat the Identity Form data
+        const beneficialOwnersFiltered = [];
+        if (this.state.hasOtherBeneficialOwners) {
+            _.each(this.state.beneficialOwners, (ownerID) => {
+                const beneficialOwner = {};
+                beneficialOwner.firstName = lodashGet(values, `beneficialOwner.${ownerID}.firstName`);
+                beneficialOwner.lastName = lodashGet(values, `beneficialOwner.${ownerID}.lastName`);
+                beneficialOwner.dob = lodashGet(values, `beneficialOwner.${ownerID}.dob`);
+                beneficialOwner.ssnLast4 = lodashGet(values, `beneficialOwner.${ownerID}.ssnLast4`);
+                beneficialOwner.street = lodashGet(values, `beneficialOwner.${ownerID}.street`);
+                beneficialOwner.city = lodashGet(values, `beneficialOwner.${ownerID}.city`);
+                beneficialOwner.state = lodashGet(values, `beneficialOwner.${ownerID}.state`);
+                beneficialOwner.zipCode = lodashGet(values, `beneficialOwner.${ownerID}.zipCode`);
+                beneficialOwnersFiltered.push(beneficialOwner);
+            });
```

Comment:
I think that we can simplify this a bit by:

1. Removing the if statement. We would still get an empty array if we iterate over an empty array. The performance gain with the if statement should be negligible here.
2. Using `_.map` to return the array instead `_.each` and manually pushing to the array
3. Defining all properties inline instead of declaring an empty object and assigning the keys

```suggestion
const beneficialOwners = _.map(this.state.beneficialOwners, ownerID => ({
    firstName: lodashGet(values, `beneficialOwner.${ownerID}.firstName`),
    lastName: lodashGet(values, `beneficialOwner.${ownerID}.lastName`),
    dob: lodashGet(values, `beneficialOwner.${ownerID}.dob`),
    ssnLast4: lodashGet(values, `beneficialOwner.${ownerID}.ssnLast4`),
    street: lodashGet(values, `beneficialOwner.${ownerID}.street`),
    city: lodashGet(values, `beneficialOwner.${ownerID}.city`),
    state: lodashGet(values, `beneficialOwner.${ownerID}.state`),
    zipCode: lodashGet(values, `beneficialOwner.${ownerID}.zipCode`),
}));
```